### ISSUE TITLE
How to use "For Each" and "For Loop" in Orchard Workflow?

### ISSUE THREAD

**Issue by SoleProgrammer** [OrchardCMS/OrchardCore#6527]
Hi,
I am looking for some documentation or tutorials about the usage of "For Each" and "For Loop" in Workflows (with examples).

Please provide me any resources you may know.
Thanks in advance.

**Comment by hishamco** [comment]
https://docs.orchardcore.net/en/dev/docs/reference/modules/Workflows/ if you didn't find what you want, please check the source, which is the valuable resource

**Comment by SoleProgrammer** [comment]
> https://docs.orchardcore.net/en/dev/docs/reference/modules/Workflows/ if you didn't find what you want, please check the source, which is the valuable resource

Thank you for your response, but I have already checked this link and as you know, none of the described activities has a link to the documentation. :|

**Comment by vahid001** [comment]
> Hi,
> I am looking for some documentation or tutorials about the usage of "For Each" and "For Loop" in Workflows (with examples).
> 
> Please provide me any resources you may know.
> Thanks in advance.

I am struggling with this issue too. Unfortunately the Orchard Document is really weak.
It looks strong with its events and tasks, but it's not worth it when we can't work with it.

Workflow has many events and tasks, but I found only one good example for using them, and that is the link below:
http://www.ideliverable.com/blog/orchard-core-workflows-walkthrough-content-approval

For most of the events and tasks such as "For Each", "For Loop", "While", "Update Content", "Delete Content", and "Retrieve Content", there is no example on the net, which makes it so hard to work with.

**Comment by vitalybrandes** [comment]
> > Hi,
> > I am looking for some documentation or tutorials about the usage of "For Each" and "For Loop" in Workflows (with examples).
> > Please provide me any resources you may know.
> > Thanks in advance.
> 
> I am struggling with this issue too. Unfortunately the Orchard Document is really weak.
> It looks strong with its events and tasks, but it's not worth it when we can't work with it.
> 
> Workflow has many events and tasks, but I found only one good example for using them, and that is the link below:
> http://www.ideliverable.com/blog/orchard-core-workflows-walkthrough-content-approval
> 
> For most of the events and tasks such as "For Each", "For Loop", "While", "Update Content", "Delete Content", and "Retrieve Content", there is no example on the net, which makes it so hard to work with.

Agree, need update content demo as well, upload file and didn't find some good explanation

**Comment by weirdyang** [comment]
> > > Hi,
> > > I am looking for some documentation or tutorials about the usage of "For Each" and "For Loop" in Workflows (with examples).
> > > Please provide me any resources you may know.
> > > Thanks in advance.
> > 
> > 
> > I am struggling with this issue too. Unfortunately the Orchard Document is really weak.
> > It looks strong with its events and tasks, but it's not worth it when we can't work with it.
> > Workflow has many events and tasks, but I found only one good example for using them, and that is the link below:
> > http://www.ideliverable.com/blog/orchard-core-workflows-walkthrough-content-approval
> > For most of the events and tasks such as "For Each", "For Loop", "While", "Update Content", "Delete Content", and "Retrieve Content", there is no example on the net, which makes it so hard to work with.
> 
> Agree, need update content demo as well, upload file and didn't find some good explanation

Check out this issue https://github.com/OrchardCMS/OrchardCore/issues/6027 the user has provided a very thorough guide on how to upload and create content

**Comment by lampersky** [comment]
I've prepared a few [recipes](https://github.com/lampersky/OrchardCoreWorkflowsExamples).
